lib/spack/spack/environment/environment.py

```diff
@@ -467,6 +468,9 @@ def from_dict(base_path, d):
 
     @property
     def _current_root(self):
+        if spack.util.atomic_update.use_renameat2():
+            return self.root
```

Comment:
See above -- I think this needs to be more careful about what *might* be there.  There aren't just two cases (symlink and renameat2), there are also cases where I used one then the other.